src/NHibernate.Test/FilterTest/DynamicFilterTest.cs

```diff
@@ -354,6 +354,35 @@ public void InStyleFilterParameter()
 			}
 		}
 
+		[Test]
+		public void InStyleFilterParameterWithHashSet()
+		{
+			//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+			// one-to-many loading tests
+			//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+			log.Info("Starting one-to-many collection loader filter tests with HashSet.");
+			using (var session = OpenSession())
+			{
+				Assert.Multiple(
+					() =>
+					{
+						session.EnableFilter("regionlist")
+						       .SetParameterList("regions", new HashSet<string> { "LA", "APAC" });
+
+						log.Debug("Performing query of Salespersons");
+						var salespersons = session.CreateQuery("from Salesperson").List();
```

Comment:
Before the fix, this case does not throw, but convert the hashset to a single string it uses as single argument in the `in` clause, causing it to fail retrieving expected results.